test/ex_doc/formatter/html_test.exs

```diff
@@ -51,7 +51,49 @@ defmodule ExDoc.Formatter.HTMLTest do
     assert File.regular?("#{output_dir}/CompiledWithDocs.html")
     assert File.regular?("#{output_dir}/CompiledWithDocs.Nested.html")
     content = File.read!("#{output_dir}/index.html")
-    assert content =~ ~r{<meta http-equiv="refresh" content="0; url=overview.html"\s*/>}
+    assert content =~ ~r{<meta http-equiv="refresh" content="0; url=overview.html">}
+  end
+
+  test "check headers for index.html and module pages" do
+    generate_docs doc_config([main: "RandomError", ])
+
+    pages = %{
+      index:  File.read!("#{output_dir}/index.html"),
+      module: File.read!("#{output_dir}/RandomError.html"),
+    }
+
+    # Regular Expressions
+    re = %{
+      shared: [
+        ~r{<meta charset="utf-8">},
+        ~r{<meta name="generator" content="ExDoc v\d+.\d+.\d+">},
+      ],
+
+      index: [
+        ~r{<title>Elixir v1.0.1 – Documentation</title>},
+        ~r{<meta name="index" content="noindex"},
+        ~r{<meta http-equiv="refresh" content="0; url=RandomError.html">},
+      ],
+
+      module: [
+        ~r{<title>RandomError – Elixir v1.0.1</title>},
+        ~r{<meta name="viewport" content="width=device-width, initial-scale=1.0">},
+        ~r{<meta http-equiv="x-ua-compatible" content="ie=edge">},
+      ],
+    }
+
+    for {page, content} <- pages do
+      for {rule_page, rules} <- re, rule_page in [:shared, page] do
+        for rule <- rules, do:
```

Comment:
I have a feeling the straight-forward version of this test would be less lines of code than the current one. And more understandable. :) Can we give it a try?